src/ServiceControl/Recoverability/Retrying/RetriesGateway.cs

```diff
@@ -41,7 +41,7 @@ public RetriesGateway(IDocumentStore store, RetryDocumentManager documentManager
 
                     if (currentBatch.Count == BatchSize)
                     {
-                        response.Add(currentBatch.ToArray());
+                        response.Add(currentBatch.Distinct().ToArray());
```

Comment:
Why not simply use HashSet with string comparer in this method?

Reply:
What is the difference?

Reply:
The currentBatch will never contain duplicates because the strings get deduped while adding. Plus we rely on HashSet dedup mechanism instead of LINQ generic distinct